src/platform/DeviceControlServer.cpp

```diff
@@ -54,12 +57,16 @@ void DeviceControlServer::CommissioningFailedTimerComplete()
 
 CHIP_ERROR DeviceControlServer::ArmFailSafe(System::Clock::Timeout expiryLength)
 {
+    mFailSafeArmed = true;
     DeviceLayer::SystemLayer().StartTimer(expiryLength, HandleArmFailSafe, this);
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR DeviceControlServer::DisarmFailSafe()
 {
+    mFailSafeArmed = false;
+    SetFabricIndex(kUndefinedFabricIndex);
+    SetNocCommandInvoked(false);
```

Comment:
This should be happening when initially arming, not when disarming...  Would be clearer.

Reply:
We should set mFailSafeArmed == false in DisarmFailSafe, right? then we should set its "have seen NOC command" boolean to false.